### ISSUE TITLE
Time between the loops keep increasing.

### ISSUE THREAD

**Issue by hrghauri** [tameemsafi/typewriterjs#35]
I am trying to write a string followed by erasing it and than writing it again in a loop. However, I have noticed the time delay between each individual loops keep increasing. The first loop takes almost no seconds. The second loops takes a bit longer, the third takes even more longer and so on... Here is the code:

```
  new Typewriter(app, { loop: true })
    .typeString("Hello World!")
    .pauseFor(250)
    .start();
```

I have managed to work around this by exploiting the callback feature. Here is the code:

```
  (function callBack() {
    new Typewriter(app)
      .typeString("Hello World!")
      .pauseFor(250)
      .deleteAll()
      .start()
      .callFunction(callBack);
  })();

```
I hope we can track down the root issue for this.

Updated: Following is the cdn I am using:
  <script src="https://unpkg.com/typewriter-effect/dist/core.js"></script>

**Comment by tameemsafi** [comment]
It should be fixed now, below is an example which will log the time it takes to finish the loop. It my differ slightly but it will be generally the same for every loop.

Example:
https://jsfiddle.net/v5uzLrpb/

**Comment by haggen** [comment]
@tameemsafi It seems to be happening again, even on the example you provided. With each loop the time between loops increase a bit.

**Comment by tameemsafi** [comment]
@haggen I have taken a look and I don't see the problem in the example above. Yes it does increase/decrease slightly but that is expected due to using `window.requestAnimationFrame`. However, it does not exponentially increase so there does not seem to be any issues. If you are experiencing an issue then please provide a way for me to reproduce the error so that I can fix it.

![image](https://user-images.githubusercontent.com/15384037/61970213-35ad7b00-afd4-11e9-83bf-30df2d25abb6.png)

**Comment by haggen** [comment]
Click (tested on Chrome 75) and open your console. https://h9ht9.csb.app/

To see the source: https://codesandbox.io/s/typewriter-effect-loop-delay-bug-h9ht9

**Comment by tameemsafi** [comment]
@haggen Thank you but it seems you are using the `console.time` function incorrectly. You need to run the `console.time("typewriter loop")` when it starts and then run `console.timeEnd("typewriter loop")` when it has completed. Please can you try that? It seems to work fine for me.

**Comment by haggen** [comment]
@tameemsafi Thank you for your quick response! :)

I'm not using it incorrectly. I'm using to measure exactly what I want to measure, which is the time *between* loops--opposed to time the whole procedure. That's what increases with each loop.

Just look at the result and you'll notice it rather easily. It:

1. Clears, writes Banana and pauses for 1 second
2. Clears, writes Apple and pauses for 1 second
3. Clears, writes Grape and pauses for 1 second
4. Starts over, but every time this step is reached it adds an additional pause that is increasingly longer each time

**Comment by tameemsafi** [comment]
@haggen I have now published `v2.10.0` please let me know if you are still seeing the issue.

**Comment by haggen** [comment]
That was fast, @tameemsafi :smile: Thank you! It looks fixed now!
